internal/pkg/service/appsproxy/proxy/apphandler/authproxy/basicauth/handler.go

```diff
@@ -74,6 +74,14 @@ func (h *Handler) CookieExpiration() time.Duration {
 	return 1 * time.Hour
 }
 
+// ServeHTTPOrError serves basic authorization pages based on these conditions:
+// nothing || cookie == nil -> render Login page GET/POST (200), no error
+// form[password=""] -> Please enter a correct password (200), error
+// form[password="b"] -> Please enter a correct password (200), error
+// form[password="a"] -> Authorized, set cookie and redirect to URL of user. (301), no error
+// cookie != nil && not signout && authorized -> Go to upstream (Based on app state), no error
+// cookie != nil && signout -> unset cookie, redirect to Login pageWriter (303), go to _proxy/SignInPath, no error
+// cookie != nil && unauthorized -> Cookie has expired (200), error
```

Comment:
I don't know whether this should be on top of module, to let someone know how the handler works(should work).